sdk/data/azappconfig/response_types.go

```diff
@@ -78,3 +79,94 @@ type SetSettingResponse struct {
 	// SyncToken contains the value returned in the Sync-Token header.
 	SyncToken SyncToken
 }
+
+// ArchiveSnapshotResponse contains the response from the ArchiveSnapshot method.
+type ArchiveSnapshotResponse struct {
+	Snapshot
+
+	// Link contains the information returned from the Link header response.
+	Link *string
+
+	// SyncToken contains the information returned from the Sync-Token header response.
+	SyncToken SyncToken
+}
+
+// RestoreSnapshotResponse contains the response from the RestoreSnapshot method.
+type RestoreSnapshotResponse struct {
+	Snapshot
+
+	// Link contains the information returned from the Link header response.
+	Link *string
+
+	// SyncToken contains the information returned from the Sync-Token header response.
+	SyncToken SyncToken
+}
+
+// NewGetSnapshotsPagerResponse contains the response from the NewGetSnapshotsPager method.
+type NewGetSnapshotsPagerResponse struct {
+	// Contains the configuration settings returned that match the setting selector provided.
+	Snapshots []Snapshot
+
+	// SyncToken contains the value returned in the Sync-Token header.
+	SyncToken SyncToken
+}
+
+// BeginCreateSnapshotResponse contains the response from the BeginCreateSnapshot method.
+type BeginCreateSnapshotResponse struct {
```

Comment:
For LRO responses, the response envelope name can omit the `Begin` prefix.

Also, there's no content in this type, is that expected? Presumably, it should return a `Snapshot` type that contains info for the newly created snapshot?

Reply:
if this is an LRO then would we want to return the Snapshot as it would not exist?

Reply:
For a `Poller[T]`, the generic type parameter is the type that's returned in the terminal success case. Here's an [example](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/security/keyvault/azadmin/backup/client.go#L37) from `azadmin`.